src/main/java/org/junit/rules/TestRuleAdapter.java

```diff
@@ -0,0 +1,75 @@
+package org.junit.rules;
+
+import org.junit.runner.Description;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+import java.lang.reflect.Method;
+
+/**
+ * A TestRuleAdapter allows to wrap {@link MethodRule} classes in
+ * {@link TestRule} classes in order to use them inside {@link RuleChain}.
+ * Assuming we have an existing method rule:
+ *
+ * <pre>
+ * class MyMethodRule implements MethodRule {
+ *     public Statement apply(final Statement base, FrameworkMethod method,
+ *                            Object target) {
+ *         return new Statement() {
+ *             &#064;Override
+ *             public void evaluate() throws Throwable {
+ *                 base.evaluate();
+ *             }
+ *         };
+ *     }
+ * }
+ * </pre>
+ *
+ * It can then be used in a {@link RuleChain} like this:
+ * <pre>
+ * class MyTest {
+ *     &#064;Rule
+ *     public final RuleChain chain = RuleChain
+ *             .outerRule(new TestRuleAdapter(new MyMethodRule(), MyTest.this))
+ *             .around(new TestRuleAdapter(new MyMethodRule(), MyTest.this));
+ *
+ *     ...
+ * }
+ * </pre>
+ *
+ * @since 4.12
+ */
+public class TestRuleAdapter implements TestRule {
```

Comment:
I am wondering if this should be package scope. What is the use case for code outside of `RuleChain` using this?